openupgrade_scripts/scripts/website_sale/15.0.1.1/post-migration.py

```diff
@@ -1,6 +1,40 @@
+import re
+
 from openupgradelib import openupgrade
 
 
+def replace_content_in_product_custom_text(env):
+    """Replace Terms and Conditions content in the new v15 template so as not to lose
+    content from previous versions if it has been customised."""
+    product_views = env["ir.ui.view"].search(
+        [("key", "=", "website_sale.product"), ("website_id", "!=", False)]
+    )
+    for view in product_views:
+        product_arch = view.arch_db
+        product_pattern = r'<p class="text-muted"(.*?)<\/p>'
+        product_matches = re.findall(product_pattern, product_arch, re.DOTALL)
+
+        if product_matches:
+            product_custom_text_view = env["ir.ui.view"].search(
+                [("key", "=", "website_sale.product_custom_text")]
+            )
+            if product_custom_text_view:
+                product_custom_text_arch = product_custom_text_view.arch_db
+                product_custom_text_pattern = (
+                    r'<p class="text-muted h6 mt-3">(.*?)<\/p>'
+                )
+                product_custom_text_matches = re.findall(
+                    product_custom_text_pattern, product_custom_text_arch, re.DOTALL
+                )
+                new_arch = product_custom_text_arch.replace(
+                    product_custom_text_matches, product_matches
+                )
+                new_product_custom_text_view = product_custom_text_view.copy()
```

Comment:
Shouldn't this only be done in case the terms are not the standard ones?

Reply:
With the change made this will only run in case the terms and conditions are not standard.
This migration script really needs to be done in v14 which is where the change is made to the templates.
You can continue to review the script in the new PR https://github.com/OCA/OpenUpgrade/pull/4096